### ISSUE TITLE
-Wtautological-constant-compare warning with clang++ 6

### ISSUE THREAD

**Issue by springmeyer** [mapbox/mapbox-gl-native#11586]
**Platform:**

OS X

**version:**

https://github.com/mapbox/mapbox-gl-native/commit/553efa38e3591ce62863d4d74222710f8e3c2337

### Steps to trigger behavior

 1. Use clang++6 from mason on OS X
 2. Build with `mkdir build && cd build && cmake ../ && make`
 3. Hit error on new warning (that I presume does not occur with clang++ 5 or earlier versions)

### Expected behavior

No warnings/no compiler errors

### Actual behavior

Hit this error:

```
/Users/dane/projects/vt-shaver-cpp/deps/mapbox-gl-native/src/mbgl/renderer/buckets/fill_extrusion_bucket.cpp:105:30: error: comparison 'const short' > 32767 is always false [-Werror,-Wtautological-constant-compare]
                    if (dist > std::numeric_limits<int16_t>::max()) {
                        ~~~~ ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Comment by jfirebaugh** [comment]
Refs #11458.

**Comment by jfirebaugh** [comment]
Fixed in #11918.
